Failure in rpc_getblockfrompeer.py

https://cirrus-ci.com/task/6199503433760768?logs=ci#L4414
https://cirrus-ci.com/task/5628523837652992?logs=ci#L4519
```bash
hash=5005fd299f78e023c07fa4cce23642e5a689c535f8adbc379825b7604b928b64)]) 
 test  2022-10-28T10:04:36.776000Z TestFramework.p2p (DEBUG): Received message from 127.0.0.1:15393: msg_sendheaders() 
 test  2022-10-28T10:04:36.776000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 139, in try_rpc
                                       fun(*args, **kwds)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/coverage.py", line 49, in __call__
                                       return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/authproxy.py", line 144, in __call__
                                       raise JSONRPCException(response['error'], status)
                                   test_framework.authproxy.JSONRPCException: Block header missing (-1)
                                   During handling of the above exception, another exception occurred:
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 133, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/rpc_getblockfrompeer.py", line 112, in run_test
                                       assert_raises_rpc_error(-1, error_msg, self.nodes[1].getblockfrompeer, blockhash, node1_interface_id)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 130, in assert_raises_rpc_error
                                       assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 147, in try_rpc
                                       message, e.error['message']))
                                   AssertionError: Expected substring not found in error message:
                                   substring: 'In prune mode, only blocks that the node has already synced previously can be fetched from a peer'
                                   error message: 'Block header missing'.
```

--- a/test/functional/rpc_getblockfrompeer.py
+++ b/test/functional/rpc_getblockfrompeer.py
@@ -100,7 +100,7 @@ class GetBlockFromPeerTest(BitcoinTestFramework):
         # Connect a P2PInterface to the pruning node and have it submit only the header of the
         # block that the pruning node has not seen
         node1_interface = self.nodes[1].add_p2p_connection(P2PInterface())
-        node1_interface.send_message(msg_headers([block]))
+        node1_interface.send_and_ping(msg_headers([block]))
 
         # Get the peer id of the P2PInterface from the pruning node
         node1_peers = self.nodes[1].getpeerinfo()

